Application/Education/ClassRegister/Diary/Frontend.php

```diff
@@ -0,0 +1,859 @@
+                            )
+                        );
+                    }
+                }
+            }
+
+            $table = new TableData($divisionTable, null, array(
+                'Year' => 'Schuljahr',
+                'Type' => 'Schulart',
+                'Division' => 'Klasse',
+                'Option' => ''
+            ), array(
+                'order' => array(
+                    array('0', 'desc'),
+                    array('1', 'asc'),
+                    array('2', 'asc'),
+                ),
+                'columnDefs' => array(
+                    array('type' => 'natural', 'targets' => 2)
+                ),
+            ));
+        }
+
+        $Stage->setContent(
+            new Layout(array(
+                new LayoutGroup(array(
+                    new LayoutRow(array(
+                        empty($buttonList)
+                            ? null
+                            : new LayoutColumn($buttonList),
+                        new LayoutColumn($table)
+                    ))
+                ), new Title(new Select() . ' Auswahl'))
+            ))
+        );
+
+        return $Stage;
+    }
+
+    /**
+     * @param Stage $Stage
+     * @param int $view
+     */
+    private function setHeaderButtonList(Stage $Stage, $view)
+    {
+        $hasTeacherRight = Access::useService()->hasAuthorization('/Education/Diary/Teacher');
+        $hasHeadmasterRight = Access::useService()->hasAuthorization('/Education/Diary/Headmaster');
+
+        $countRights = 0;
+        if ($hasTeacherRight) {
+            $countRights++;
+        }
+        if ($hasHeadmasterRight) {
+            $countRights++;
+        }
+
+        if ($countRights > 1) {
+            if ($hasTeacherRight) {
+                if ($view == View::TEACHER) {
+                    $Stage->addButton(new Standard(new Info(new Bold('Ansicht: Lehrer')),
+                        self::BASE_ROUTE . '/Teacher', new Edit()));
+                } else {
+                    $Stage->addButton(new Standard('Ansicht: Lehrer',
+                        self::BASE_ROUTE . '/Teacher'));
+                }
+            }
+            if ($hasHeadmasterRight) {
+                if ($view == View::HEADMASTER) {
+                    $Stage->addButton(new Standard(new Info(new Bold('Ansicht: Leitung')),
+                        self::BASE_ROUTE . '/Headmaster', new Edit()));
+                } else {
+                    $Stage->addButton(new Standard('Ansicht: Leitung',
+                        self::BASE_ROUTE . '/Headmaster'));
+                }
+            }
+        }
+    }
+
+    /**
+     * @param null $DivisionId
+     * @param null $GroupId
+     * @param string $BasicRoute
+     * @param bool $StudentId
+     *
+     * @return Stage|string
+     */
+    public function frontendDiary(
+        $DivisionId = null,
+        $GroupId = null,
+        $BasicRoute = '/Education/Diary/Teacher',
+        $StudentId = null
```

Comment:
Codacy found an issue: [Incorrect spacing between argument "$StudentId" and equals sign; expected 0 but found 1](https://app.codacy.com/manual/gerdchristian-kunze/SPHERE-Framework/pullRequest?prid=5757982)